### ISSUE TITLE
[ally]: Combobox Default & with Layer - VO focus order

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon-website#3552]
## Environment

> Operating system

> Browser

> Automated testing tool and ruleset

> Assistive technology used to verify

## Detailed description

> What version of the Carbon Design System are you using?

> What did you expect to happen?
1. Focus should be set back on the triggering control after selection
2. Redundant swipe focus should be removed

> What happened instead?
Following behavior is observed in VO iPhone. 
1. After selecting an option from the dropdown, the Screen reader focus is not set back on the triggering control. 
2. After selection, When the user tries to navigate back, it takes 5 blank swipe stopped to reach the triggering control. Note: This happens each time when the user selects an option.

> What WCAG 2.1 checkpoint does the issue violate?

## Steps to reproduce the issue

Default: https://react.carbondesignsystem.com/?path=/story/components-combobox--default
With layer: https://react.carbondesignsystem.com/?path=/story/components-combobox--with-layer

> Please create a reduced test case in CodeSandbox
> https://codesandbox.io/s/github/IBM/carbon-components-react/tree/master/examples/codesandbox

## Additional information

- Screenshots or code
- Notes

**Comment by Huixh111** [comment]
ii
